File: Java/Topico_03/src/TestaCirculo.java
import java.util.*;

class Circulo {
    private int x, y, raio;
    public static final double PI = 3.14159;

    public Circulo(int x, int y, int raio) {
        this.x = x;
        this.y = y;
        this.raio = raio;
    }

    public int getX() { 
        return x; 
    }

    public int getY() { 
        return y; 
    }

    public int getRaio() { 
        return raio;
    }

    public void setRaio(int raio) {
        this.raio = raio;
    }

    public double circunferencia() {
        return 2 * PI * raio;
    }
}

public class TestaCirculo {
    public static void main(String[] args) {
        Circulo circulo1, circulo2, circulo3;
        Scanner input = new Scanner(System.in);

        int x1 = input.nextInt();
        int y1 = input.nextInt();
        int raio1 = input.nextInt();
        int x2 = input.nextInt();
        int y2 = input.nextInt();
        int raio2 = input.nextInt();
        int x3 = input.nextInt();
        int y3 = input.nextInt();
        int raio3 = input.nextInt();
        int novoRaio1 = input.nextInt();
        int novoRaio2 = input.nextInt();
        int novoRaio3 = input.nextInt();
        
        circulo1 = new Circulo(x1, y1, raio1);
        circulo2 = new Circulo(x2, y2, raio2);
        circulo3 = new Circulo(x3, y3, raio3);
        
        Circulo[] vetor1 = {circulo1, circulo2, circulo3};                                  //1
        Circulo[] vetor2 = vetor1;                                                          //2 e 3
        Circulo[] vetor3 = new Circulo[3];                                                  //4
        vetor3[0] = new Circulo(vetor1[0].getX(), vetor1[0].getY(), vetor1[0].getRaio());   //5
        vetor3[1] = new Circulo(vetor1[1].getX(), vetor1[1].getY(), vetor1[1].getRaio());   //5
        vetor3[2] = new Circulo(vetor1[2].getX(), vetor1[2].getY(), vetor1[2].getRaio());   //5


        vetor1[0].setRaio(novoRaio1);                                                       //6
        vetor1[1].setRaio(novoRaio2);                                                       //6
        vetor1[2].setRaio(novoRaio3);                                                       //6
        
        System.out.println("vetor1:");
        for (int i = 0; i < 3; i++) {
            System.out.printf("%d %d %d\n", vetor1[i].getX(), vetor1[i].getY(), vetor1[i].getRaio());
        }
        
        System.out.println("vetor2:");
        for (int i = 0; i < 3; i++) {
            System.out.printf("%d %d %d\n", vetor2[i].getX(), vetor2[i].getY(), vetor2[i].getRaio());
        }        
        
        System.out.println("vetor3:");
        for (int i = 0; i < 3; i++) {
            System.out.printf("%d %d %d\n", vetor3[i].getX(), vetor3[i].getY(), vetor3[i].getRaio());
        }
        input.close();
    }
}

File: C++/testaCirculo.cpp
#include <iostream>
#include <vector>
using namespace std;

class Circulo {
    private:
        int x, y, raio;
    public:
        const double PI = 3.14159;
        Circulo(int x, int y, int raio) {
            this->x = x;
            this->y = y;
            this->raio = raio;
        }
        Circulo() {
            this->x = 0;
            this->y = 0;
            this->raio = 0;
        }
        int getX() {
            return x;
        }
        int getY() {
            return y;
        }
        int getRaio() {
            return raio;
        }
        void setRaio(int raio) {
            this->raio = raio;
        }
        double circunferencia() {
            return 2 * PI * raio;
        }
};

int main() {
    int x1, y1, raio1, x2, y2, raio2, x3, y3, raio3, novoRaio1, novoRaio2, novoRaio3;
    cin >> x1 >> y1 >> raio1;
    cin >> x2 >> y2 >> raio2;
    cin >> x3 >> y3 >> raio3;
    cin >> novoRaio1;
    cin >> novoRaio2;
    cin >> novoRaio3;

    Circulo circulo1(x1, y1, raio1);
    Circulo circulo2(x2, y2, raio2);
    Circulo circulo3(x3, y3, raio3);

    vector<Circulo> vetor1 = {circulo1, circulo2, circulo3};
    vector<Circulo> vetor2 = vetor1;
    vector<Circulo> vetor3;
    vetor3.push_back(circulo1);
    vetor3.push_back(circulo2);
    vetor3.push_back(circulo3);
    
    vetor1[0].setRaio(novoRaio1);
    vetor1[1].setRaio(novoRaio2);
    vetor1[2].setRaio(novoRaio3);

    cout << "vetor1:" << endl;
    for(int i = 0; i < 3; i++) {
        cout << vetor1[i].getX() << " " << vetor1[i].getY() << " " << vetor1[i].getRaio() << endl;
    }
    
    cout << "vetor2:" << endl;
    for(int i = 0; i < 3; i++) {
        cout << vetor2[i].getX() << " " << vetor2[i].getY() << " " << vetor2[i].getRaio() << endl;
    } 
    
    cout << "vetor3:" << endl;
    for(int i = 0; i < 3; i++) {
        cout << vetor3[i].getX() << " " << vetor3[i].getY() << " " << vetor3[i].getRaio() << endl;
    }
    return 0;
}